src/back/hlsl/writer.rs

```diff
@@ -43,13 +43,13 @@ pub struct Writer<'a, W> {
 }
 
 impl<'a, W: Write> Writer<'a, W> {
-    pub fn new(out: W, options: &'a Options) -> Self {
+    pub fn new(out: W, options: &'a Options, module: &Module) -> Self {
         Self {
             out,
             names: crate::FastHashMap::default(),
             namer: proc::Namer::default(),
             options,
-            ep_inputs: Vec::with_capacity(3),
+            ep_inputs: Vec::with_capacity(module.entry_points.len()),
```

Comment:
let's not pass the `module` just for this. Instead, just use `Vec::new` and expect this to be re-used